File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
      s = set(nums)
      max_len = 0
      i = 0
      while(i < len(nums)):
        n = nums[i]
        c = 1
        while n+1 in s:
            n+=1
            c+=1
        max_len = max(max_len, c)
        i+=1

      return max_len







        

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.java
class Solution {

    public int lengthOfLIS(int[] nums) {

        return dfs(nums, 0, Integer.MIN_VALUE);  
    }

private int dfs(int[] nums, int i, int j) {
    if (i >= nums.length)
        return 0;
    int include = 0;
    if(nums[i] > j)
        include = 1 + dfs(nums, i + 1, nums[i]); 
    int exclude = dfs(nums, i + 1, j);
    return Math.max(include, exclude);  
}
}
